lib/src/core/implementation/signature_body_source_parameters.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:typed_data';
+
+/// Parameters required for constructing a signature body source.
+///
+/// This class encapsulates the parameters required for constructing a signature body source,
+/// including the HTTP method, URI, headers, nonce, timestamp, client ID, components of the signature body source,
+/// and optional request body.
+class SignatureBodySourceParameters {
+  /// The HTTP method of the request.
+  final String method;
+
+  /// The URI of the request.
+  final Uri uri;
+
+  /// The headers of the request.
+  final Map<String, String> headers;
+
+  /// The nonce value for the request.
+  final String nonce;
+
+  /// The timestamp of the request.
+  final int timestamp;
+
+  /// The client ID associated with the request.
+  final String clientId;
+
+  /// The components of the signature body source.
+  final List<String> signatureBodySourceComponents;
+
+  /// The optional request body.
+  ///
+  /// If provided, it should be a list of byte lists.
```

Comment:
So is this a list of list?
Shouldn't it simply be a `Uint8List`?

I suggest you also bring the unit tests of the Core library in this PR. It should help validating that the code works.